test/factories/member.factory.ts

```diff
@@ -19,14 +19,22 @@ function BaseAccountFactory<T extends AccountType>(baseAccount: Partial<Account>
   };
 }
 
-export const MemberFactory = (m: Partial<Member> = {}) => ({
-  email: faker.internet.email().toLowerCase(),
-  extra: faker.helpers.arrayElement([{ lang: faker.helpers.arrayElement(['en', 'fr', 'de']) }, {}]),
-  enableSaveActions: m.enableSaveActions ?? true,
-  isValidated: m.isValidated ?? true,
-  ...BaseAccountFactory({ type: AccountType.Individual }),
-  ...m,
-});
+export const MemberFactory = (m: Partial<Member> = {}): Member => {
+  const isValidated = m.isValidated ?? true;
+  return {
+    email: faker.internet.email().toLowerCase(),
+    extra: faker.helpers.arrayElement([
+      { lang: faker.helpers.arrayElement(['en', 'fr', 'de']) },
+      {},
+    ]),
+    enableSaveActions: m.enableSaveActions ?? true,
+    isValidated,
+    lastAuthenticatedAt: isValidated ? (m.lastAuthenticatedAt ?? faker.date.anytime()) : null,
+    userAgreementsDate: m.userAgreementsDate ?? faker.date.anytime(),
+    ...BaseAccountFactory({ type: AccountType.Individual }),
+    ...m,
+  } as Member;
```

Comment:
Why is the cast necessary ? maybe add a comment so we know when we can remove it.